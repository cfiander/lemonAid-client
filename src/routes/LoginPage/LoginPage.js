import React, { Component } from 'react'
import LoginForm from '../../components/LoginForm/LoginForm'

export default class LoginPage extends Component {
  static defaultProps = {
    location: {},
    history: {
      push: () => {},
    },
  }

  handleLoginSuccess = () => {
    const { location, history } = this.props
    const destination = (location.state || {}).from || '/'
    history.push(destination)
  }

  render() {
    return (
        <section className="login">
        <h1 className="heroTitle">Login</h1>
        <LoginForm
          onLoginSuccess={this.handleLoginSuccess}
        />
        </section>
    )
  }
}
