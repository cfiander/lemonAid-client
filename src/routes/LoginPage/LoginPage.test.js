
import React from 'react';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import LoginPage from './LoginPage';

describe(`IngredientsPage component`, () => {
  it('renders an IngredientsPage by default', () => {
    const wrapper = shallow(<LoginPage
         />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})