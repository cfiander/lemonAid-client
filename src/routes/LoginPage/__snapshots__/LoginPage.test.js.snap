// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IngredientsPage component renders an IngredientsPage by default 1`] = `
<section>
  <h2>
    Login
  </h2>
  <LoginForm
    onLoginSuccess={[Function]}
  />
</section>
`;
